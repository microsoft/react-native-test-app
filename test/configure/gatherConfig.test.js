//
// Copyright (c) Microsoft Corporation
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the root directory of this source tree.
//
// @ts-check

describe("gatherConfig()", () => {
  const { mockParams } = require("./mockParams");
  const { gatherConfig } = require("../../scripts/configure");

  test("returns configuration for all platforms", () => {
    expect(gatherConfig(mockParams())).toMatchSnapshot();
  });

  test("returns configuration for a single platform", () => {
    expect(gatherConfig(mockParams({ platforms: ["ios"] }))).toMatchSnapshot();
  });

  test("returns configuration for arbitrary platforms", () => {
    const params = mockParams({ platforms: ["android", "ios"] });
    expect(gatherConfig(params)).toMatchSnapshot();
  });

  test("flattens configuration for a single platform only", () => {
    const iosOnly = mockParams({ platforms: ["ios"], flatten: true });
    expect(gatherConfig(iosOnly)).toMatchSnapshot();

    const mobile = mockParams({ platforms: ["android", "ios"], flatten: true });
    expect(gatherConfig(mobile)).toMatchSnapshot();
  });
});
