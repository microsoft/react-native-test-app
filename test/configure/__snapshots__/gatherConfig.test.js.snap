// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatherConfig() flattens configuration for a single platform only 1`] = `
Object {
  "dependencies": Object {},
  "files": Object {
    ".gitignore": Object {
      "source": "example/.gitignore",
    },
    ".watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "Podfile": "require_relative '../test_app'

use_flipper!({ 'Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3' })

workspace 'Test.xcworkspace'

use_test_app!
",
    "babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "react-native.config.js": "const project = (() => {
  try {
    const { iosProjectPath } = require(\\"react-native-test-app\\");
    const project = iosProjectPath(\\".\\");
    return project
      ? {
          ios: {
            project,
          },
        }
      : undefined;
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
  },
  "oldFiles": Array [
    "Podfile.lock",
    "Pods",
    "Test.xcodeproj",
    "Test.xcworkspace",
  ],
  "scripts": Object {
    "build:ios": "mkdirp dist && react-native bundle --entry-file index.js --platform ios --dev true --bundle-output dist/main.ios.jsbundle --assets-dest dist",
    "ios": "react-native run-ios --project-path .",
    "start": "react-native start",
  },
}
`;

exports[`gatherConfig() flattens configuration for a single platform only 2`] = `
Object {
  "dependencies": Object {},
  "files": Object {
    ".gitignore": Object {
      "source": "example/.gitignore",
    },
    ".watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "android/build.gradle": "buildscript {
    def androidTestAppDir = \\"../../android\\"
    apply(from: \\"\${androidTestAppDir}/dependencies.gradle\\")
    apply(from: \\"\${androidTestAppDir}/test-app-util.gradle\\")

    repositories {
        mavenCentral()
        google()
    }

    dependencies {
        classpath(\\"com.android.tools.build:gradle:\${androidPluginVersion}\\")

        if (isNewArchitectureEnabled(project)) {
            classpath(\\"com.facebook.react:react-native-gradle-plugin\\")
            classpath(\\"de.undercouch:gradle-download-task:5.1.0\\")
        }
    }
}

allprojects {
    repositories {
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(\\"\${rootDir}/../react-native/android\\")
        }
        mavenCentral()
        google()
    }
}
",
    "android/gradle.properties": Object {
      "source": "example/android/gradle.properties",
    },
    "android/gradle/wrapper/gradle-wrapper.jar": Object {
      "source": "example/android/gradle/wrapper/gradle-wrapper.jar",
    },
    "android/gradle/wrapper/gradle-wrapper.properties": Object {
      "source": "example/android/gradle/wrapper/gradle-wrapper.properties",
    },
    "android/gradlew": Object {
      "source": "example/android/gradlew",
    },
    "android/gradlew.bat": Object {
      "source": "example/android/gradlew.bat",
    },
    "android/settings.gradle": "pluginManagement {
    repositories {
        gradlePluginPortal()
        mavenCentral()
        google()
    }
}

rootProject.name = \\"Test\\"

apply(from: \\"../../test-app.gradle\\")
applyTestAppSettings(settings)
",
    "babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "ios/Podfile": "require_relative '../../test_app'

use_flipper!({ 'Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3' })

workspace 'Test.xcworkspace'

use_test_app!
",
    "metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "react-native.config.js": "const project = (() => {
  const fs = require(\\"fs\\");
  const path = require(\\"path\\");
  try {
    const {
      androidManifestPath,
      iosProjectPath,
      windowsProjectPath,
    } = require(\\"react-native-test-app\\");
    const iosProject = iosProjectPath(\\"ios\\");
    return {
      android: {
        sourceDir: \\"android\\",
        manifestPath: androidManifestPath(path.join(__dirname, \\"android\\")),
      },
      windows: fs.existsSync(\\"windows/Test.sln\\") && {
        sourceDir: \\"windows\\",
        solutionFile: \\"Test.sln\\",
        project: windowsProjectPath(path.join(__dirname, \\"windows\\")),
      },
      ...(iosProject ? { ios: { project: iosProject } } : undefined),
    };
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
  },
  "oldFiles": Array [
    "ios/Podfile.lock",
    "ios/Pods",
    "ios/Test.xcodeproj",
    "ios/Test.xcworkspace",
  ],
  "scripts": Object {
    "android": "react-native run-android",
    "build:android": "mkdirp dist/res && react-native bundle --entry-file index.js --platform android --dev true --bundle-output dist/main.android.jsbundle --assets-dest dist/res",
    "build:ios": "mkdirp dist && react-native bundle --entry-file index.js --platform ios --dev true --bundle-output dist/main.ios.jsbundle --assets-dest dist",
    "ios": "react-native run-ios",
    "start": "react-native start",
  },
}
`;

exports[`gatherConfig() returns common configuration 1`] = `
Object {
  "dependencies": Object {},
  "files": Object {
    ".gitignore": Object {
      "source": "example/.gitignore",
    },
    ".watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "common/.gitignore": Object {
      "source": "example/.gitignore",
    },
    "common/.watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "common/babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "common/metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "common/react-native.config.js": "const project = (() => {
  const fs = require(\\"fs\\");
  const path = require(\\"path\\");
  try {
    const {
      androidManifestPath,
      iosProjectPath,
      windowsProjectPath,
    } = require(\\"react-native-test-app\\");
    const iosProject = iosProjectPath(\\"ios\\");
    return {
      android: {
        sourceDir: \\"android\\",
        manifestPath: androidManifestPath(path.join(__dirname, \\"android\\")),
      },
      windows: fs.existsSync(\\"windows/Test.sln\\") && {
        sourceDir: \\"windows\\",
        solutionFile: \\"Test.sln\\",
        project: windowsProjectPath(path.join(__dirname, \\"windows\\")),
      },
      ...(iosProject ? { ios: { project: iosProject } } : undefined),
    };
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
    "metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "react-native.config.js": "const project = (() => {
  const fs = require(\\"fs\\");
  const path = require(\\"path\\");
  try {
    const {
      androidManifestPath,
      iosProjectPath,
      windowsProjectPath,
    } = require(\\"react-native-test-app\\");
    const iosProject = iosProjectPath(\\"ios\\");
    return {
      android: {
        sourceDir: \\"android\\",
        manifestPath: androidManifestPath(path.join(__dirname, \\"android\\")),
      },
      windows: fs.existsSync(\\"windows/Test.sln\\") && {
        sourceDir: \\"windows\\",
        solutionFile: \\"Test.sln\\",
        project: windowsProjectPath(path.join(__dirname, \\"windows\\")),
      },
      ...(iosProject ? { ios: { project: iosProject } } : undefined),
    };
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
  },
  "oldFiles": Array [],
  "scripts": Object {
    "start": "react-native start",
  },
}
`;

exports[`gatherConfig() returns configuration for a single platform 1`] = `
Object {
  "dependencies": Object {},
  "files": Object {
    ".gitignore": Object {
      "source": "example/.gitignore",
    },
    ".watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "ios/Podfile": "require_relative '../../test_app'

use_flipper!({ 'Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3' })

workspace 'Test.xcworkspace'

use_test_app!
",
    "metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "react-native.config.js": "const project = (() => {
  const fs = require(\\"fs\\");
  const path = require(\\"path\\");
  try {
    const {
      androidManifestPath,
      iosProjectPath,
      windowsProjectPath,
    } = require(\\"react-native-test-app\\");
    const iosProject = iosProjectPath(\\"ios\\");
    return {
      android: {
        sourceDir: \\"android\\",
        manifestPath: androidManifestPath(path.join(__dirname, \\"android\\")),
      },
      windows: fs.existsSync(\\"windows/Test.sln\\") && {
        sourceDir: \\"windows\\",
        solutionFile: \\"Test.sln\\",
        project: windowsProjectPath(path.join(__dirname, \\"windows\\")),
      },
      ...(iosProject ? { ios: { project: iosProject } } : undefined),
    };
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
  },
  "oldFiles": Array [
    "ios/Podfile.lock",
    "ios/Pods",
    "ios/Test.xcodeproj",
    "ios/Test.xcworkspace",
  ],
  "scripts": Object {
    "build:ios": "mkdirp dist && react-native bundle --entry-file index.js --platform ios --dev true --bundle-output dist/main.ios.jsbundle --assets-dest dist",
    "ios": "react-native run-ios",
    "start": "react-native start",
  },
}
`;

exports[`gatherConfig() returns configuration for all platforms 1`] = `
Object {
  "dependencies": Object {
    "react-native-macos": "^0.63.0",
    "react-native-windows": "^0.63.0",
  },
  "files": Object {
    ".gitignore": Object {
      "source": "example/.gitignore",
    },
    ".watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "android/build.gradle": "buildscript {
    def androidTestAppDir = \\"../../android\\"
    apply(from: \\"\${androidTestAppDir}/dependencies.gradle\\")
    apply(from: \\"\${androidTestAppDir}/test-app-util.gradle\\")

    repositories {
        mavenCentral()
        google()
    }

    dependencies {
        classpath(\\"com.android.tools.build:gradle:\${androidPluginVersion}\\")

        if (isNewArchitectureEnabled(project)) {
            classpath(\\"com.facebook.react:react-native-gradle-plugin\\")
            classpath(\\"de.undercouch:gradle-download-task:5.1.0\\")
        }
    }
}

allprojects {
    repositories {
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(\\"\${rootDir}/../react-native/android\\")
        }
        mavenCentral()
        google()
    }
}
",
    "android/gradle.properties": Object {
      "source": "example/android/gradle.properties",
    },
    "android/gradle/wrapper/gradle-wrapper.jar": Object {
      "source": "example/android/gradle/wrapper/gradle-wrapper.jar",
    },
    "android/gradle/wrapper/gradle-wrapper.properties": Object {
      "source": "example/android/gradle/wrapper/gradle-wrapper.properties",
    },
    "android/gradlew": Object {
      "source": "example/android/gradlew",
    },
    "android/gradlew.bat": Object {
      "source": "example/android/gradlew.bat",
    },
    "android/settings.gradle": "pluginManagement {
    repositories {
        gradlePluginPortal()
        mavenCentral()
        google()
    }
}

rootProject.name = \\"Test\\"

apply(from: \\"../../test-app.gradle\\")
applyTestAppSettings(settings)
",
    "babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "ios/Podfile": "require_relative '../../test_app'

use_flipper!({ 'Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3' })

workspace 'Test.xcworkspace'

use_test_app!
",
    "macos/Podfile": "require_relative '../../macos/test_app'

workspace 'Test.xcworkspace'

use_test_app!
",
    "metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "react-native.config.js": "const project = (() => {
  const fs = require(\\"fs\\");
  const path = require(\\"path\\");
  try {
    const {
      androidManifestPath,
      iosProjectPath,
      windowsProjectPath,
    } = require(\\"react-native-test-app\\");
    const iosProject = iosProjectPath(\\"ios\\");
    return {
      android: {
        sourceDir: \\"android\\",
        manifestPath: androidManifestPath(path.join(__dirname, \\"android\\")),
      },
      windows: fs.existsSync(\\"windows/Test.sln\\") && {
        sourceDir: \\"windows\\",
        solutionFile: \\"Test.sln\\",
        project: windowsProjectPath(path.join(__dirname, \\"windows\\")),
      },
      ...(iosProject ? { ios: { project: iosProject } } : undefined),
    };
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
    "windows/.gitignore": Object {
      "source": "example/windows/.gitignore",
    },
  },
  "oldFiles": Array [
    "ios/Podfile.lock",
    "ios/Pods",
    "ios/Test.xcodeproj",
    "ios/Test.xcworkspace",
    "macos/Podfile.lock",
    "macos/Pods",
    "macos/Test.xcodeproj",
    "macos/Test.xcworkspace",
    "windows/Test.sln",
    "windows/Test.vcxproj",
    "windows/Test/Test.vcxproj",
  ],
  "scripts": Object {
    "android": "react-native run-android",
    "build:android": "mkdirp dist/res && react-native bundle --entry-file index.js --platform android --dev true --bundle-output dist/main.android.jsbundle --assets-dest dist/res",
    "build:ios": "mkdirp dist && react-native bundle --entry-file index.js --platform ios --dev true --bundle-output dist/main.ios.jsbundle --assets-dest dist",
    "build:macos": "mkdirp dist && react-native bundle --entry-file index.js --platform macos --dev true --bundle-output dist/main.macos.jsbundle --assets-dest dist",
    "build:windows": "mkdirp dist && react-native bundle --entry-file index.js --platform windows --dev true --bundle-output dist/main.windows.bundle --assets-dest dist",
    "ios": "react-native run-ios",
    "macos": "react-native run-macos --scheme Test",
    "start": "react-native start",
    "windows": "react-native run-windows --sln windows/Test.sln",
  },
}
`;

exports[`gatherConfig() returns configuration for arbitrary platforms 1`] = `
Object {
  "dependencies": Object {},
  "files": Object {
    ".gitignore": Object {
      "source": "example/.gitignore",
    },
    ".watchmanconfig": Object {
      "source": "node_modules/react-native/template/_watchmanconfig",
    },
    "android/build.gradle": "buildscript {
    def androidTestAppDir = \\"../../android\\"
    apply(from: \\"\${androidTestAppDir}/dependencies.gradle\\")
    apply(from: \\"\${androidTestAppDir}/test-app-util.gradle\\")

    repositories {
        mavenCentral()
        google()
    }

    dependencies {
        classpath(\\"com.android.tools.build:gradle:\${androidPluginVersion}\\")

        if (isNewArchitectureEnabled(project)) {
            classpath(\\"com.facebook.react:react-native-gradle-plugin\\")
            classpath(\\"de.undercouch:gradle-download-task:5.1.0\\")
        }
    }
}

allprojects {
    repositories {
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url(\\"\${rootDir}/../react-native/android\\")
        }
        mavenCentral()
        google()
    }
}
",
    "android/gradle.properties": Object {
      "source": "example/android/gradle.properties",
    },
    "android/gradle/wrapper/gradle-wrapper.jar": Object {
      "source": "example/android/gradle/wrapper/gradle-wrapper.jar",
    },
    "android/gradle/wrapper/gradle-wrapper.properties": Object {
      "source": "example/android/gradle/wrapper/gradle-wrapper.properties",
    },
    "android/gradlew": Object {
      "source": "example/android/gradlew",
    },
    "android/gradlew.bat": Object {
      "source": "example/android/gradlew.bat",
    },
    "android/settings.gradle": "pluginManagement {
    repositories {
        gradlePluginPortal()
        mavenCentral()
        google()
    }
}

rootProject.name = \\"Test\\"

apply(from: \\"../../test-app.gradle\\")
applyTestAppSettings(settings)
",
    "babel.config.js": Object {
      "source": "node_modules/react-native/template/babel.config.js",
    },
    "ios/Podfile": "require_relative '../../test_app'

use_flipper!({ 'Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3' })

workspace 'Test.xcworkspace'

use_test_app!
",
    "metro.config.js": Object {
      "source": "example/metro.config.js",
    },
    "react-native.config.js": "const project = (() => {
  const fs = require(\\"fs\\");
  const path = require(\\"path\\");
  try {
    const {
      androidManifestPath,
      iosProjectPath,
      windowsProjectPath,
    } = require(\\"react-native-test-app\\");
    const iosProject = iosProjectPath(\\"ios\\");
    return {
      android: {
        sourceDir: \\"android\\",
        manifestPath: androidManifestPath(path.join(__dirname, \\"android\\")),
      },
      windows: fs.existsSync(\\"windows/Test.sln\\") && {
        sourceDir: \\"windows\\",
        solutionFile: \\"Test.sln\\",
        project: windowsProjectPath(path.join(__dirname, \\"windows\\")),
      },
      ...(iosProject ? { ios: { project: iosProject } } : undefined),
    };
  } catch (_) {
    return undefined;
  }
})();

module.exports = {
  ...(project ? { project } : undefined),
};
",
  },
  "oldFiles": Array [
    "ios/Podfile.lock",
    "ios/Pods",
    "ios/Test.xcodeproj",
    "ios/Test.xcworkspace",
  ],
  "scripts": Object {
    "android": "react-native run-android",
    "build:android": "mkdirp dist/res && react-native bundle --entry-file index.js --platform android --dev true --bundle-output dist/main.android.jsbundle --assets-dest dist/res",
    "build:ios": "mkdirp dist && react-native bundle --entry-file index.js --platform ios --dev true --bundle-output dist/main.ios.jsbundle --assets-dest dist",
    "ios": "react-native run-ios",
    "start": "react-native start",
  },
}
`;
